woocommerce/woocommerce
# Make sure the `woocommerce` page hook name remains untranslated

### All Submissions:






### Changes proposed in this Pull Request:

Due to [a WP bug](https://core.trac.wordpress.org/ticket/18857), if a menu item's title is translated, page hooks and screen IDs associated to its subpages are constructed using the translated name (despite what the WP docs say).

This means WC itself (and 3rd party) code can't safely rely on `load-woocommerce_page_(subpage)` hooks to perform actions or assume `woocommerce_page_(subpage)` is the screen ID of a given subpage.

We already have code that assumes the above to perform actions or enqueue scripts/styles, so until WP addresses the issue, we can add a tiny workaround to our codebase, which is precisely what this PR does.

Closes #35677.






### How to test the changes in this Pull Request:



1. Make sure you're on `trunk`.
2. Install the "Say What?" plugin (slug: `say-what`) via Plugins > Add New.
3. Go to Tools > Text Changes.
4. Click "Add New".
5. Fill out the form like this:
   <img width="691" alt="Screenshot 2022-11-23 at 11 01 32" src="https://user-images.githubusercontent.com/184724/203592621-db99ae5f-c2ef-45ee-9197-616d64fa3e96.png">
6. Click "Add".
7. Make sure the main "WooCommerce" menu now reads "LaWooCommerce".
8. Go to WC > Settings > Advanced > Features and enable the "high performance order storage" feature.
9. Go to WC > Settings > Advanced > Custom data stores, and make sure "WooCommerce orders tables" is the datastore in use.
10. Go to WC > Orders.
11. You'll get a fatal error (as described in #35677).
12. Check out this branch.
13. Refresh the page. No error should be triggered.



### Testing (alternative)

1. Add the following PHP snippet to a suitable location:
   ```php
   add_action(
	   'admin_menu',
	   function() {
		   add_submenu_page(
			   'woocommerce',
			   'My Page',
			   'My Page',
			   'manage_options',
			   'my-page',
			   function() {
				   var_dump( get_current_screen()->id );
			   }
		   );
	   }
   );
   ```
2. Click the "My Page" menu item under WooCommerce.
3. Follow steps 2-7 above to translate the main WooCommerce menu.
4. On `trunk` you'll see that the screen ID includes the translated slug.
5. On this branch, the screen ID remains not translated.


### Other information:




### FOR PR REVIEWER ONLY:
